src/services/DynamoDBService.ts

```diff
@@ -174,4 +174,39 @@ export class DynamoDBService {
 
         return Promise.all(promiseBatch);
     }
+
+    /**
+     * Performs a write transaction on the specified table.
+     * @param item - the item to be inserted or updated during the transaciton.
+     */
+    public transactWrite(item: any): Promise<PromiseResult<DocumentClient.TransactWriteItemsOutput, AWS.AWSError>> {
+        const query: DocumentClient.TransactWriteItemsInput = { TransactItems: [
+            {
+                Put: {
+                    TableName: this.tableName,
+                    Item: item,
+                    ConditionExpression: "testNumber <> :testNumberVal",
```

Comment:
Why do you need this condition? To not allow the lambda to re-insert an object with the same testNumber?

Reply:
As discussed, this logic has been changed